src/commands/new.rs

```diff
@@ -28,21 +44,39 @@ impl NewCommand {
         self.run_template(repo, id).await
     }
 
+    async fn list_templates(&self) -> anyhow::Result<()> {
+        let entries = hub_api::index().await.unwrap();
+        let matches = entries.iter()
+            .filter(|e| self.is_terms_match(e))
+            .sorted_by_key(|e| e.title())
+            .collect_vec();
+
+        if matches.is_empty() {
+            println!("No templates match your search terms");
+            return Ok(());
+        }
+
+        for entry in matches {
+            println!("Template: {}\nDescription: {}\n", entry.title(), entry.summary());
+        }
+
+        Ok(())
+    }
+
     async fn run_template(&self, repo: String, id: String) -> anyhow::Result<()> {
         use spin_templates::*;
 
-        let tempdir = tempfile::tempdir().unwrap();
-        let template_manager = spin_templates::TemplateManager::in_dir(tempdir.path());
+        let manager = TemplateManager::try_default()?;
 
         let source = TemplateSource::try_from_git(&repo, &None, &crate::spin::version())?;
         let options = InstallOptions::default();
-        template_manager.install(&source, &options, &DiscardingProgressReporter).await?;
+        manager.install(&source, &options, &DiscardingProgressReporter).await?;
 
-        let template = template_manager.get(&id).unwrap().unwrap();
+        let template = manager.get(&id).unwrap().unwrap();
         let options = RunOptions {
             variant: TemplateVariantInfo::NewApplication,
-            name: self.name.clone(),
-            output_path: PathBuf::from(&self.name), // TODO: path safe,
+            name: self.name.clone().unwrap(), 
```

Comment:
Nit: capture the validated or prompted name into a non-Option so you don't need to unwrap here.